### DISCUSSION TITLE
Retrying NFS mount - NFS over TCP not available

### DISCUSSION THREAD

**Discussion by Stu619** [ltsp/ltsp#401] [Q&A]
Just loaded latest LTSP on Ubuntu 20.04. Client booting (pxe) goes fine until I get the above-noted error. Assistance would be appreciated. I'm new to this version of LTSP but ran a 30 person office of LTSP linux desktops 20 years ago.

**Comment by alkisg** [comment]
Hi, did you run `sudo ltsp nfs` on your server?
What's the output of `showmount -e localhost` on the server?

**Reply by Stu619** [reply]
Thanks, I hadn't run sudo ltsp nfs. showmount -e localhost says "clnt_create: RPC: Program not registered".

**Comment by alkisg** [comment]
Yeah that would cause the issue; please have another look at the instructions in https://ltsp.org/docs/installation/ and make sure that you've run all of the related commands.

**Reply by Stu619** [reply]
Thanks. Not sure how I ran all the other commands except the nfs one.

**Comment by Stu619** [comment]
Still having the same issue. I have 2 network cards in my machine but I am only using one. Could that be the problem?

**Comment by alkisg** [comment]
No, it shouldn't be a problem. Do you have a firewall?
Make sure that `showmount -e localhost` on the server shows the NFS shares,
and also that `showmount -e server-ip` from any other PC in the same subnet, with the nfs-common package installed, also shows the NFS shares.

**Reply by Stu619** [reply]
I think it's working now. For some reason nfs was failing because /srv/nfs4 and /srv/nfs4/homes didn't exist. I created those and now nfs seems to start properly.

**Reply by Stu619** [reply]
This is what I now get from showmount -e localhost

showmount -e localhost
Export list for localhost:
/srv/nfs4/homes gss/krb5i
/srv/nfs4       gss/krb5i
/srv/tftp/ltsp  *
/srv/ltsp       *

**Comment by alkisg** [comment]
Usually homes are mounted over SSHFS, so there's no need for an /srv/nfs4 directory; LTSP doesn't handle that.
But if you already have nfs4 and kerberos etc configured, that's fine too.

**Reply by Stu619** [reply]
I'm not purposefully using nfs for that. Is sshfs the default and if so, why would mine be defaulting to nfs?

**Comment by alkisg** [comment]
The `ltsp nfs` command configures /srv/ltsp and /srv/tftp/ltsp as NFSv3 exports in order for the clients to boot from them.
Then, by default, the clients use SSHFS to connect to their homes.

We do not use or configure NFSv4 in LTSP, but if you want to manually configure it, you may.
If you do not, then just undo your manual changes; `ltsp nfs` does the right thing, there's no need for you to edit any configuration files or to manually create any directories.

To see your current NFS configuration, one command is:

```shell
grep -rEv '^#|^$' /etc/exports*
```

LTSP creates /etc/exports.d/ltsp-nfs.exports; anything else that you may see in the output was created by you and isn't needed by LTSP.

**Reply by Stu619** [reply]
I originally said I was using Ubuntu 20.04 but I forgot I upgraded to 20.10, would that be the reason I've got NFSv4? My nfs server wouldn't start without /srv/nfs4 and /srv/nfs4/homes. Maybe those were just needed to start the nfs server? I didn't put those in there.

**Comment by alkisg** [comment]
No, LTSP doesn't need, configure or use NFS4.
If you want me to guess about how this might have happened unrelated to LTSP, paste the output of the grep command that I mentioned.
